crates/service_utils/src/helpers.rs

```diff
@@ -401,3 +411,91 @@ pub fn parse_config_tags(
         }
     }
 }
+
+pub async fn execute_webhook_call<T>(
+    webhook_object: Value,
+    payload: T,
+    config_version_opt: &Option<String>,
+    event: WebhookEvent,
+) -> Result<(), AppError>
+where
+    T: Serialize,
+{
+    match serde_json::from_value::<Webhook>(webhook_object) {
+        Ok(webhook) => {
+            let http_client = reqwest::Client::new();
+            let mut header_array = webhook
+                .headers
+                .into_iter()
+                .filter_map(|key| match key {
+                    HeadersEnum::ConfigVersion => {
+                        if let Some(config_version) = config_version_opt {
+                            Some((
+                                "x-config-version".to_string(),
+                                config_version.to_string(),
+                            ))
+                        } else {
+                            None
+                        }
+                    }
+                })
+                .collect::<Vec<(String, String)>>();
+
+            if let Ok(auth) =
+                serde_json::from_value::<Authorization>(webhook.authorization)
+            {
+                let auth_token_value: String =
+                    get_from_env_unsafe(&auth.value).map_err(|err| {
+                        log::error!("Failed to retrieve authentication token for the webhook with error: {}", err);
+                        AppError::WebhookError(
+                            format!("Failed to retrieve authentication token for the webhook. Please verify the credentials in TenantConfig.")
+                        )
+                    })?;
+                header_array.push((auth.key, auth_token_value));
+            }
+
+            let mut headers = HeaderMap::new();
+            header_array.iter().for_each(|(name, value)| {
```

Comment:
I think, we had a function for doing this, can you please verify once
most probably `construct_request_headers` from `service_utils` should be the one

Reply:
Already had tried this, but construct_request_headers takes &[&str, &str]